### ISSUE TITLE
Cloned document does not set new Owner

### ISSUE THREAD

**Issue by campersau** [AngleSharp/AngleSharp#514]
After cloning a document the Owner property of the elements of the cloned document are pointing to the original document and not to the cloned document.

Here is a simple testcase to demonstrate the issue:

```csharp
        [Test]
        public void CloneDocumentSetsNewOwner()
        {
            const string html = "<div>document</div>";

            var parser = new HtmlParser();
            var originalDocument = parser.Parse(html);
            var newDocument = (IDocument)originalDocument.Clone(true);

            var div = newDocument.QuerySelector("div");
            Assert.AreSame(newDocument, div.Owner);

            div.TextContent = "cloned document";
            var newHtml = div.Owner.DocumentElement.OuterHtml;
            Assert.True(newHtml.Contains("cloned document"));            
        }
```

In the browser this works:

```js
var newDocument = document.cloneNode(true);
var div = newDocument.querySelector("div");
div.ownerDocument === newDocument; // true
```

**Comment by campersau** [comment]
Okay, so according to the spec this might not be a bug.

> [And, cloning Document, DocumentType, Entity, and Notation nodes is implementation dependent.](https://www.w3.org/TR/DOM-Level-2-Core/core.html#ID-3A0ED0A4)

**Comment by FlorianRappl** [comment]
Actually, this is a bug in my opinion. I do not set the new owner (I copy all original properties, i.e., I also copy the original owner). Will be changed with the v0.10!

Thanks for the report 👍 .

**Comment by FlorianRappl** [comment]
Fixed, see PR #515.
